koku/masu/api/manifest/views.py

```diff
@@ -149,3 +162,101 @@ def get_one_manifest_file(self, request, *args, **kwargs):
             raise ManifestException("Invalid manifest id or report status id")
         queryset = UsageReportStatusSerializer(queryset, many=True).data
         return Response(queryset)
+
+
+def manifest_failed_filter(queryset, name, value):
+    """A custom filter to return manfests that did or did not fail based on the boolean value of failed."""
+    filters = {}
+    for step in ManifestStep:
+        # if the failed field IS NULL then the manifest did not fail so we have to inverse the search criteria.
+        filters[f"state__{step}__{ManifestState.FAILED}__isnull"] = not value
+    return queryset.filter(**filters)
+
+
+def manifest_running_filter(queryset, name, value):
+    """A custom filter to return manfests that are running tasks."""
+    for step in ManifestStep:
+        queryset = queryset.filter(**{f"state__{step}__{ManifestState.END}__isnull": value})
+    return queryset
+
+
+class ManifestFilter(FilterSet):
+    """Custom Manifest filters."""
+
+    account_id = CharListFilter(
+        field_name="provider__customer__account_id", lookup_expr="provider__customer__account_id__icontains"
+    )
+    org_id = CharListFilter(
+        field_name="provider__customer__org_id", lookup_expr="provider__customer__org_id__icontains"
+    )
+    schema_name = CharListFilter(
+        field_name="provider__customer__schema_name", lookup_expr="provider__customer__schema_name__icontains"
+    )
+    created = DateFromToRangeFilter(field_name="creation_datetime")
+    updated = DateFromToRangeFilter(field_name="manifest_updated_datetime")
+    completed = DateFromToRangeFilter(field_name="completed_datetime")
+    manifest_failed = BooleanFilter(method=manifest_failed_filter)
+    manifest_running = BooleanFilter(method=manifest_running_filter)
+
+    class Meta:
+        model = CostUsageReportManifest
+        fields = [
+            "provider",
+            "account_id",
+            "org_id",
+            "schema_name",
+            "created",
+            "updated",
+            "completed",
+            "manifest_failed",
+            "manifest_running",
```

Comment:
[Optional]: Is it worth changing these to just be `failed` and `running` since we already know its relating to the manifest via the endpoint, thus its already inferred. For example. `v1/manifests/?failed=True`

Reply:
Yeah, I like this change, updated to just `failed` and `running`.